cpp/src/arrow/filesystem/azurefs_test.cc

```diff
@@ -423,8 +423,11 @@ class TestGeneric : public ::testing::Test, public GenericFileSystemTest {
 class TestAzuriteGeneric : public TestGeneric {
  public:
   void SetUp() override {
-    ASSERT_OK_AND_ASSIGN(auto env, AzuriteEnv::GetInstance());
-    SetUpInternal(env);
+    auto env_result = AzuriteEnv::GetInstance();
+    if (!env_result.ok()) {
+      GTEST_SKIP() << "Failed to setup: " << env_result.status().ToString();
```

Comment:
couldn't this makes us miss tests being skipped due to issues with our environment? We could potentially stop testing Azure and not realize

Reply:
Thanks @kou I think I prefer the second approach so at least we have the safeguard of the macOS cpp job that installs azurite to fail if the `install_azurite.sh` script starts silently failing again.

Reply:
OK. I've changed to use the approach.

Reply:
AzureFS tests are excluded On Ubuntu 22.04:

https://github.com/apache/arrow/actions/runs/18025582047/job/51291868066?pr=47623#step:7:6958

```text
+ ctest --label-regex unittest --output-on-failure --parallel 4 --repeat until-pass:3 --timeout 300 --exclude-regex arrow-azurefs-test
```

AzureFS tests are executed On Ubuntu 24.04:

https://github.com/apache/arrow/actions/runs/18025582047/job/51291868086?pr=47623#step:7:7607

```text
        Start  76: arrow-azurefs-test
```